endpoint/runcode.py

```diff
@@ -71,7 +71,14 @@ def on_post(self, req, resp, id):
             if user.is_org():
                 req.context['result']['report'] = reporter.report
 
-            execution.report = reporter.report
+            # truncate too long reports so that they fit into MEDIUMTEXT database field
+            # (MEDIUMTEXT is ~16MiB, but UTF encoding may take more space than 1B per character)
+            report_text = reporter.report
+            report_text_safe_length = 10*(1024**2)
+            if len(report_text) > report_text_safe_length:
```

Comment:
Nevidím přesně typ report_text, ale nešlo by použít něco takového?

```suggestion
            if len(report_text.encode('utf-8')) > report_text_safe_length:
```

Reply:
Fajn nápad! Ale pokud se budeme pohybovat hodně pod velikostí `MEDIUMTEXT`, tak si nemyslím, že je potřeba dělat encoding, nástřel z délky stringu mi přijde rychlejší a dostatečná

Reply:
Osobně bych ho tam dal. Teď víme že by s tím mohl být problém, ale časem bychom to mohli zapomenout. Velmi drobný overhead je za mě lepší než pak potenciálně několik hodin debugování, než znova přijdeme na to, že tohle je možná problém.
Pokud bys to ale chtěl vynechat, taky to jde, výše je aspoň ten komentář.